homeassistant/components/unifi/light.py

```diff
@@ -0,0 +1,196 @@
+        if ATTR_RGB_COLOR in kwargs
+        else device.led_override_color
+    )
+
+    await hub.api.request(
+        DeviceSetLedStatus.create(
+            device=device,
+            status=status,
+            brightness=brightness,
+            color=color,
+        )
+    )
+
+
+def _parse_hex_color(
+    hex_color: str | None, default: tuple[int, int, int]
+) -> tuple[int, int, int]:
+    """Parse hex color string to RGB tuple."""
+    if not hex_color:
+        return default
+
+    color_hex = hex_color.lstrip("#")
+    if len(color_hex) != 6:
+        return default
+
+    try:
+        rgb_values = [int(color_hex[i : i + 2], 16) for i in (0, 2, 4)]
+        return (rgb_values[0], rgb_values[1], rgb_values[2])
+    except ValueError:
+        return default
+
+
+@dataclass(frozen=True, kw_only=True)
+class UnifiLightEntityDescription[HandlerT: APIHandler, ApiItemT: ApiItem](
+    LightEntityDescription, UnifiEntityDescription[HandlerT, ApiItemT]
+):
+    """Class describing UniFi light entity."""
+
+    control_fn: Callable[..., Awaitable[None]]
+    is_on_fn: Callable[[UnifiHub, ApiItemT], bool]
+
+
+ENTITY_DESCRIPTIONS: tuple[UnifiLightEntityDescription, ...] = (
+    UnifiLightEntityDescription[Devices, Device](
+        key="LED control",
+        translation_key="led_control",
+        allowed_fn=lambda hub, obj_id: True,
+        api_handler_fn=lambda api: api.devices,
+        available_fn=async_device_available_fn,
+        control_fn=async_device_led_control_fn,
+        device_info_fn=async_device_device_info_fn,
+        is_on_fn=async_device_led_is_on_fn,
+        name_fn=lambda device: "LED",
+        object_fn=lambda api, obj_id: api.devices[obj_id],
+        supported_fn=async_device_led_supported_fn,
+        unique_id_fn=lambda hub, obj_id: f"led-{obj_id}",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: UnifiConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up lights for UniFi Network integration."""
+    config_entry.runtime_data.entity_loader.register_platform(
+        async_add_entities,
+        UnifiLightEntity,
+        ENTITY_DESCRIPTIONS,
+    )
+
+
+class UnifiLightEntity[HandlerT: APIHandler, ApiItemT: ApiItem](
+    UnifiEntity[HandlerT, ApiItemT], LightEntity
+):
+    """Base representation of a UniFi light."""
+
+    entity_description: UnifiLightEntityDescription[HandlerT, ApiItemT]
+
+    def __init__(
+        self,
+        obj_id: str,
+        hub: UnifiHub,
+        description: UnifiLightEntityDescription[HandlerT, ApiItemT],
+    ) -> None:
+        """Initialize UniFi light entity."""
+        super().__init__(obj_id, hub, description)
+        self._attr_supported_features = LightEntityFeature(0)
+        self._attr_color_mode = ColorMode.RGB
+        self._attr_supported_color_modes = {ColorMode.RGB}
```

Comment:
The states are all static so could be done without init, unless `       _attr_supported_color_modes = {ColorMode.RGB}` is illegal to do outside of init. Anyway all the static ones should be defined on class level and only init what is needed within init

```suggestion
    entity_description: UnifiLightEntityDescription[HandlerT, ApiItemT]

        _attr_supported_features = LightEntityFeature(0)
        _attr_color_mode = ColorMode.RGB
        _attr_supported_color_modes = {ColorMode.RGB}
```